golem-worker-service-base/src/worker_bridge_execution/mod.rs

```diff
@@ -77,6 +78,28 @@ impl WorkerRequest {
             function_params.push(json);
         }
 
+        let idempotency_key = if let Some(expr) = &resolved_route
+            .resolved_worker_binding_template
+            .idempotency_key
+        {
+            let idempotency_key_value = expr
+                .evaluate(&resolved_route.typed_value_from_input)
+                .map_err(|err| err.to_string())?;
+
+            let idempotency_key = match idempotency_key_value {
+                TypeAnnotatedValue::Str(value) => value,
+                _ => return Err("Idempotency Key is not a string".to_string()),
+            };
+
+            Some(IdempotencyKey::new(idempotency_key))
+        } else {
+            resolved_route
+                .headers
+                .get("idempotency-key")
```

Comment:
Is this working case-insensitively?

Reply:
headers should not be case sensitive, i added test